app/controllers/concerns/api/contract_period_filterable.rb

```diff
@@ -7,5 +7,15 @@ module ContractPeriodFilterable
     def contract_period_years
       params.dig(:filter, :cohort)
     end
+
+    def contract_period
+      ContractPeriod.find_by(year: contract_period_years)
+    end
+
+    def contract_periods
+      return ContractPeriod.where("year > 2020") if contract_period_years.blank?
```

Comment:
can we use the constant `ECF_FIRST_YEAR`?

Reply:
I realised it's used in statements query but we forgot about it 😅  but you're right not in schools